pkg/machines/components/networks/networkOverviewTab.jsx

```diff
@@ -0,0 +1,171 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { networkId } from '../../helpers.js';
+import { WarningAlert } from '../notification/inlineNotification.jsx';
+import cockpit from 'cockpit';
+
+import './networkOverviewTab.css';
+import 'form-layout.less';
+
+const _ = cockpit.gettext;
+
+const DHCPHost = (host, index, family, idPrefix) => {
+    const id = `${idPrefix}-${family}-dhcp-host-${index}`;
+
+    let hostVals = [];
+    if (host.name)
+        hostVals.push(_("Name: ") + host.name);
+    if (host.mac) // MAC for ipv4, ID for ipv6
+        hostVals.push("MAC: " + host.mac);
+    else if (host.id)
+        hostVals.push("ID: " + host.id);
+    if (host.ip)
+        hostVals.push("IP: " + host.ip);
+
+    const hostInfo = hostVals.join(", ");
+
+    return (<React.Fragment key={index}>
+        <label className='control-label' htmlFor={id}> {`DHCP host ${index + 1}:`} </label>
+        <div id={id}> {hostInfo} </div>
+    </React.Fragment>);
+};
+
+export const NetworkOverviewTab = ({ network, actionError, actionErrorDetail, onActionErrorDismiss }) => {
+    const idPrefix = `${networkId(network.name, network.connectionName)}`;
+
+    let qosInboundVals = [];
+    if (network.bandwidth.inbound && network.bandwidth.inbound.average)
+        qosInboundVals.push(_("Average: ") + network.bandwidth.inbound.average);
+    if (network.bandwidth.inbound && network.bandwidth.inbound.burst)
+        qosInboundVals.push(_("Burst: ") + network.bandwidth.inbound.burst);
+    if (network.bandwidth.inbound && network.bandwidth.inbound.peak)
+        qosInboundVals.push(_("Peak: ") + network.bandwidth.inbound.peak);
+
+    let qosOutboundVals = [];
+    if (network.bandwidth.outbound && network.bandwidth.outbound.average)
+        qosOutboundVals.push(_("Average: ") + network.bandwidth.outbound.average);
+    if (network.bandwidth.outbound && network.bandwidth.outbound.burst)
+        qosOutboundVals.push(_("Burst: ") + network.bandwidth.outbound.burst);
+    if (network.bandwidth.outbound && network.bandwidth.outbound.peak)
+        qosOutboundVals.push(_("Peak: ") + network.bandwidth.outbound.peak);
+
+    const qosInbound = qosInboundVals.join(", ");
+    const qosOutbound = qosOutboundVals.join(", ");
+
+    let ip = [];
+    // Libvirt allows network to have multiple ipv6 and ipv4 addresses.
+    // But we only first one of each
+    ip[0] = network.ip.find(ip => ip.family === "ipv4");
+    ip[1] = network.ip.find(ip => ip.family === "ipv6");
+
+    return (
+        <div className="networks-page-grid">
+            { actionError && <WarningAlert text={actionError} textId={`${idPrefix}-error`} detail={actionErrorDetail} onDismiss={onActionErrorDismiss} /> }
+            <div className='ct-form-layout'>
+                <hr />
```

Comment:
How come you're adding a HR element here?

The alignment of the headers are a bit off, and maybe this is why.
![screenshot from 2019-02-20 14-19-49](https://user-images.githubusercontent.com/209895/53094552-bbeffd80-351a-11e9-8160-067c712b4695.png)

Reply:
And the spacing between the rows seems to be different between the two columns. Not a big deal really, but might be nice to fix, just so it looks proper and professional.

Reply:
I think this is done by the ct-form-layout div wrapper. It tries to get the full space inside the grid it has. But I aggree. I think @garrett wanted to address something similar in the package updates page, in history, when the package items are taking the whole space when they are too few.

Reply:
About spacing, yeah that is cause by ct-form-layour. It tries to strech both columns to same height.

And HR element is there to add some space between groups (like between General and QoS, or betweem IPv4 and IPv6 address). Without HR, it would look like this:
![screenshot from 2019-02-20 14-39-32](https://user-images.githubusercontent.com/42733240/53095656-b0ea9c80-351d-11e9-88f4-22f0a0aa2db2.png)
So if you like it more withou HR, it can change it.

Reply:
Also I don't know what dou you mean by headers. If you mean labels like General, QoS, IPv6 Address..., it's because i align them to left, while labels of properties like Persistent, Autostart, Inbound... are aligned to right. Do you want them all to the right?

Reply:
I meant the vertical alignment so that General and IPv4 address have equal spacing from the top.
![screenshot from 2019-02-20 16-25-21](https://user-images.githubusercontent.com/209895/53102599-5147bd80-352c-11e9-9475-a1dc6e2ba3d9.png)

Reply:
I'm OK with keeping the HR at the top. But the headers need to line up vertically at the top of the tab. So it might be an issue with ct-form-layout indeed. Do you know what's going on here, @garrett ?

Reply:
Hmm, I just realised that it doesn't make sense to have HR above top header (creates useless free space divider above top headers). It's only useful to have HR if there is another "group" above. So I removed HR only for top headers, and top headers line up now. I don't know if that's sufficient solution for you. However problem with different row spacing persists (may be caused be cause by ct-form-layout) persists.
![screenshot from 2019-02-18 16-13-11](https://user-images.githubusercontent.com/42733240/53103441-f1eaad00-352d-11e9-82fb-a45930b21b45.png)

Reply:
Hmm, maybe problem was not in ct-form-layout, but in the grid I use. So I just told it to align items to top by `align-items: start;`, and spacing is now consistend, and top header are lined up :) 
![screenshot from 2019-02-20 17-02-42](https://user-images.githubusercontent.com/42733240/53105500-886c9d80-3531-11e9-8507-923e2639bfac.png)

Reply:
What do you thing @andreasn ?

Reply:
> About spacing between rows, yeah that is cause by ct-form-layour. It tries to strech both columns to the same height.

I made a fix for this. It's... somewhere... in a PR... perhaps merged, perhaps not. I see it locally in git commit 4ac76856f69d7438e359b103a94f651210f3ba02... but I'm not sure which branches it is included in.

Reply:
OK, cherry-picked the commit and it's now f6542182173233de9ae4106aaf89d569ee07b589 in this branch.

Reply:
FWIW: It was an issue in either the container grid or in the form-layout grid... but I decided to fix it in form-layout with `align-self`, to prevent manual changes each and every time. :wink:

Reply:
Thanks @garrett. Your fix aligns items correctly, just as shown in the picture I posted above.
So i think that resolves all the comments @andreasn had for design :).

Reply:
Whenever possible, it's nicer to fix things across Cockpit versus locally. I'm happy you can remove the code and rely on the fixes in form-layout instead. :grin: